File: t-db.py
"""
Test MongoDB connection
"""
import sys
from db import connect

print('Connecting to MongoDB...')
try:
    client = connect()
    # Test connection with ping
    client.admin.command('ping')
    print('Connected to MongoDB ✓')
    
    # Get connection info
    connection_info = client.address
    server = connection_info[0] if connection_info else "unknown"
    
    print(f'Server: {server}')
    client.close()
    print('Connection closed ✓')
    sys.exit(0)
except Exception as e:
    print('Connection failed ✗')
    print(f'Error: {str(e)}')
    sys.exit(1) 

File: t-db.js
// Test MongoDB connection
import test from 'node:test';
import assert from 'node:assert';
import connectDB from './db.js';

test('MongoDB connection', async () => {
  console.log('Connecting to MongoDB...');
  
  // Connect to MongoDB
  const client = await connectDB();
  
  try {
    // Test connection with ping
    const ping = await client.db('admin').command({ ping: 1 });
    assert.strictEqual(ping.ok, 1, 'MongoDB connection failed');
    
    // Log success info
    console.log('Connected to MongoDB ✓');
    console.log(`Server: ${client.options.hosts[0]}`);
  } finally {
    // Always close connection
    await client.close();
    console.log('Connection closed ✓');
  }
}); 